components/UserProfile.tsx

```diff
@@ -0,0 +1,26 @@
+import Image from "next/image";
+import { FolderData } from "@/types/commonTypes";
+import styles from "./UserProfile.module.css";
+
+function UserProfile({ folderData }: { folderData: FolderData | {} }) {
+  const owner = folderData && (folderData as FolderData).owner;
+  const profileImageSource = owner && owner.profileImageSource;
+  const profileName = owner && owner.name;
+  return (
+    <div className={styles.container}>
+      {profileImageSource && (
+        <Image
+          width={256}
+          height={256}
+          className={styles.img}
+          src={profileImageSource}
+          alt=""
+        />
+      )}
+      {profileName && <div className={styles.name}>{profileName}</div>}
+      <div className={styles.folderName}>{(folderData as FolderData).name}</div>
+    </div>
+  );
+}
```

Comment:
초기 렌더링 시 folderData를 받아오지 않아 빈 객체로 전달되는 경우에 대한 조건 처리를 값을 사용하는 부분마다 해주셨는데,
아직 forderData가 전달되지 않은 경우에는 한꺼번에 early return하는 방식으로 구현하는 게 더 좋을 것 같습니다. 그러면 이후 코드에서는 folderData에 대한 타입이 명확해지기 때문에 일일이 조건을 달지 않아도 됩니다!

```suggestion
function UserProfile({ folderData }: { folderData: FolderData }) {
  if(!folderData) return null;
  
  ...
}
```
혹은 상위 컴포넌트에서 동일한 처리를 할 수도 있겠네요! 차이점에 대해서는 한번 생각해보시면 좋을 것 같습니다.

Reply:
또 다른 컴포넌트에서는 props의 타입을 따로 interface로 선언해 주셨는데, 이런 부분들도 통일해주는 게 좋습니다.